YouTrackAutomation/Functions/New-YTProject.ps1

```diff
@@ -0,0 +1,92 @@
+
+function New-YTProject
+{
+    <#
+    .SYNOPSIS
+    Creates a new project in YouTrack.
+
+    .DESCRIPTION
+    The `New-YTProject` function creates a new project in YouTrack. The function requires the following parameters:
+
+    * `Name`: The name of the project.
+    * `ShortName`: The short name of the project.
+    * `Leader`: The id or the name of the project owner.
+
+    .EXAMPLE
+    New-YTProject -Session $session -Name 'Demo Project' -ShortName 'DEMO' -Leader 'admin'
+
+    Demonstrates creating a new project in YouTrack with the name `Demo Project`, the short name `DEMO`, and the project
+    owner `admin`.
+
+    .EXAMPLE
+    New-YTProject -Session $session -Name 'Demo Project' -ShortName 'DEMO' -Leader '2-1'
+
+    Demonstrates creating a new project in YouTrack with the name `Demo Project`, the short name `DEMO`, and the project
+    owner `admin`, but using the project owner's id instead of their name.
+    #>
+    [CmdletBinding()]
+    param(
+        # The session object for a YouTrack Session. Create a new Session using `New-XWSession`.
+        [Parameter(Mandatory)]
+        [Object] $Session,
+
+        # The name of the project.
+        [Parameter(Mandatory)]
+        [String] $Name,
+
+        # The short name of the project.
+        [Parameter(Mandatory)]
+        [String] $ShortName,
+
+        # The id or the name of the project owner.
+        [Parameter(Mandatory)]
+        [String] $Leader,

```

Comment:
We want to make as few web requests as possible, so either change this to LeaderID or allow users to pass a leader object, short name, or ID. For the second object, change its type to Object then create a Resolve-YTLeader function that handles that.